Document/0x04f-Testing-Network-Communication.md

```diff
@@ -108,14 +108,150 @@ If that's the case, you are now able to see the complete network traffic that is
+
+For all major Linux and Unix operating systems you need:
+- hostapd,
+- dnsmasq,
+- iptables,
+- wpa_supplicant,
+- airmon-ng or
+- its equivalents.
+
+For Kali Linux you can install these tools with `apt-get`:
+
+```shell
+$ apt-get update
+$ apt-get install hostapd dnsmasq aircrack-ng
+```
+> iptables and wpa_supplicant are installed by default on Kali Linux.
+
+You should check if your one of your Wifi card has AP capabilities. This can be done by using the command `iwconfig` on Kali Linux:
+
+    ```shell
+    $ iw list | grep AP 
+    ```
+
+If your Wifi card has AP capabilities, you can use it to create Access Point.
+
+##### Configuration
+
+We focus on configuration files for Kali Linux. Following values were defined:
+- wlan1 - id of the AP network interface (with AP capabilities),
+- wlan0 - id of the target network interface (this can be wired interface or other Wifi card)
+- 10.0.0.0/24 - IP addresses and mask of AP network
+
+You need following configuration files:
+
+- hostapd.conf
+
+    ```
+    # Name of the WiFi interface we use 
+    interface=wlan1
+    # Use the nl80211 driver
+    driver=nl80211
+    hw_mode=g
+    channel=6
+    wmm_enabled=1
+    macaddr_acl=0
+    auth_algs=1
+    ignore_broadcast_ssid=0
+    wpa=2
+    wpa_key_mgmt=WPA-PSK
+    rsn_pairwise=CCMP
+    # Name of the AP network
+    ssid=STM-AP
+    # Password of the AP network
+    wpa_passphrase=password
+    ```
+
+- wpa_supplicant.conf
+
+    ```
+    network={
+        ssid="NAME_OF_THE_TARGET_NETWORK"
+        psk="PASSWORD_OF_THE_TARGET_NETWORK"
+    }
+    ```
+
+- dnsmasq.conf
+
+    ```
+    interface=wlan1
+    dhcp-range=10.0.0.10,10.0.0.250,12h
+    dhcp-option=3,10.0.0.1
+    dhcp-option=6,10.0.0.1
+    server=8.8.8.8
+    log-queries
+    log-dhcp
+    listen-address=127.0.0.1
+    ```
+
+> These are example configs. Some adjustments may be required.
+
+##### MITM Attack
+
+To be able to get a man-in-the-middle position you need to run above configuration. This can be done by using following commands on Kali Linux:
+
+    ```shell
+    # check if other process is not using Wifi interfaces
+    $ airmon-ng check kill
+    # configure IP address of the AP network interface
+    $ ifconfig wlan1 10.0.0.1 up
+    # start access point
+    $ hostapd hostapd.conf
+    # connect the target network interface
+    $ wpa_supplicant -B -i wlan0 -c wpa_supplicant.conf
+    # run DNS server
+    $ dnsmasq -C dnsmasq.conf -d
+    # enable routing 
+    $ echo 1 > /proc/sys/net/ipv4/ip_forward
+    # iptables will NAT connections from AP network interface to the target network interface
+    $ iptables --flush
+    $ iptables --table nat --append POSTROUTING --out-interface wlan0 -j MASQUERADE
+    $ iptables --append FORWARD --in-interface wlan1 -j ACCEPT 
+    $ iptables -t nat -A POSTROUTING -j MASQUERADE
+    ```
+
+Now you can connect mobile devices to the Access Point.
+
+##### Network Analyzer Tool
```

Comment:
Make this #### instead of #####, then it's a separate paragraph, applicable to both scnearios above.

Reply:
Done.